presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardOrganizerUtil.java

```diff
@@ -0,0 +1,190 @@
+        Set<ShardMetadata> shardMetadatas = shardManager.getNodeShards("node1");
+
+        Set<ShardIndexInfo> expected = shardInfoToShardIndexInfo(tableInfo, shards, false);
+        Set<ShardIndexInfo> actual = ImmutableSet.copyOf((toShardIndexInfo(dbi, metadataDao, tableInfo, shardMetadatas, false)));
+        assertEquals(actual, expected);
+
+        Set<ShardIndexInfo> expectedSortRange = shardInfoToShardIndexInfo(tableInfo, shards, true);
+        Set<ShardIndexInfo> actualSortRange = ImmutableSet.copyOf((toShardIndexInfo(dbi, metadataDao, tableInfo, shardMetadatas, true)));
+        assertEquals(actualSortRange, expectedSortRange);
+    }
+
+    private Set<ShardIndexInfo> shardInfoToShardIndexInfo(Table tableInfo, Collection<ShardInfo> shards, boolean includeSort)
+    {
+        List<TableColumn> sortColumns = metadataDao.listSortColumns(tableInfo.getTableId());
+        List<Type> sortTypes = sortColumns.stream().map(TableColumn::getDataType).collect(Collectors.toList());
+        ImmutableSet.Builder<ShardIndexInfo> indexInfos = ImmutableSet.builder();
+        for (ShardInfo shard : shards) {
+            Map<Long, ColumnStats> stats = Maps.uniqueIndex(shard.getColumnStats(), ColumnStats::getColumnId);
+
+            Optional<ShardRange> shardRange = Optional.empty();
+            if (includeSort) {
+                ImmutableList.Builder<Object> minValues = ImmutableList.builder();
+                ImmutableList.Builder<Object> maxValues = ImmutableList.builder();
+                for (TableColumn column : sortColumns) {
+                    ColumnStats columnStats = stats.get(column.getColumnId());
+                    minValues.add(columnStats.getMin());
+                    maxValues.add(columnStats.getMax());
+                }
+                shardRange = Optional.of(ShardRange.of(new Tuple(sortTypes, minValues.build()), new Tuple(sortTypes, maxValues.build())));
+            }
+
+            Optional<ShardRange> temporalRange = Optional.empty();
+            if (tableInfo.getTemporalColumnId().isPresent()) {
+                ColumnStats columnStats = stats.get(tableInfo.getTemporalColumnId().getAsLong());
+                Type type = metadataDao.getTableColumn(tableInfo.getTableId(), columnStats.getColumnId()).getDataType();
+                temporalRange = Optional.of(ShardRange.of(new Tuple(type, columnStats.getMin()), new Tuple(type, columnStats.getMax())));
+            }
+            indexInfos.add(new ShardIndexInfo(
+                    tableInfo.getTableId(),
+                    OptionalInt.empty(),
+                    shard.getShardUuid(),
+                    shard.getRowCount(),
+                    shard.getUncompressedSize(),
+                    shardRange,
+                    temporalRange));
+        }
+        return indexInfos.build();
+    }
+
+    private long createTable(String name, List<ColumnInfo> columns)
```

Comment:
Rather than replicating the logic here, we should use `RaptorMetadata`. See `TestShardMetadataRecordCursor` for an example of doing this in a test